pandas/core/internals/managers.py

```diff
@@ -970,27 +970,37 @@ def fast_xs(self, loc: int) -> SingleBlockManager:
         immutable_ea = isinstance(dtype, ExtensionDtype) and dtype._is_immutable
 
         if isinstance(dtype, ExtensionDtype) and not immutable_ea:
-            result = self.blocks[0].iget((slice(None), loc))
-            if isinstance(result, np.ndarray):
-                empty_fill_value = result[0]
-            else:
-                empty_fill_value = result._values_for_factorize()[0][0]
             cls = dtype.construct_array_type()
-            result = cls._empty((n,), dtype=dtype, fill_value=empty_fill_value)
         else:
-            # error: Argument "dtype" to "empty" has incompatible type
-            # "Union[Type[object], dtype[Any], ExtensionDtype, None]"; expected
-            # "None"
-            result = np.empty(
-                n, dtype=object if immutable_ea else dtype  # type: ignore[arg-type]
+            cls = None
+
+        if hasattr(cls, "_from_scalars"):
```

Comment:
Just realized that this hasattr check is redundant for EAs if we have _from_scalars in the base class.  Will clean up next commit.